kernel/msg_q.c

```diff
@@ -69,8 +69,8 @@ u32_t _handler_k_msgq_init(u32_t q, u32_t buffer, u32_t msg_size,
 {
 	_SYSCALL_ARG4;
 
-	_SYSCALL_IS_OBJ(q, K_OBJ_MSGQ, 1, ssf);
-	_SYSCALL_MEMORY(buffer, msg_size * max_msgs, 1, ssf);
+	_SYSCALL_OBJ_INIT(q, K_OBJ_MSGQ, ssf);
+	_SYSCALL_MEMORY_WRITE(buffer, msg_size * max_msgs, ssf);
```

Comment:
Unrelated to the patch, but if `msg_size * max_msgs` overflows, since they're both unsigned, this check can be bypassed.  Better to use `__builtin_mul_overflow()`, maybe wrapped in a `_SYSCALL_MEMORY_WRITE_ARRAY()` macro.

Reply:
There are other instances where a multiplication is performed, which could lead to the same problem.

Reply:
i'll add this to this PR good catch